janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -0,0 +1,349 @@
+import org.janusgraph.diskstorage.keycolumnvalue.StandardStoreFeatures;
+import org.janusgraph.diskstorage.keycolumnvalue.StoreFeatures;
+import org.janusgraph.diskstorage.keycolumnvalue.StoreTransaction;
+
+import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.BatchStatement.Type;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Cluster.Builder;
+import com.datastax.driver.core.JdkSSLOptions;
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.policies.DCAwareRoundRobinPolicy;
+import com.datastax.driver.core.policies.TokenAwarePolicy;
+
+import javaslang.Tuple;
+import javaslang.collection.Array;
+import javaslang.collection.HashMap;
+import javaslang.collection.Iterator;
+import javaslang.collection.Seq;
+import javaslang.concurrent.Future;
+import javaslang.control.Option;
+
+public class CQLStoreManager extends DistributedStoreManager implements KeyColumnValueStoreManager {
+
+    private static final int DEFAULT_PORT = 9042;
+
+    private final String keyspace;
+    private final boolean atomicBatch;
+    private final Cluster cluster;
+    private final Session session;
+    private final StoreFeatures storeFeatures;
+    private final Map<String, CQLKeyColumnValueStore> openStores;
+
+    public CQLStoreManager(final Configuration configuration) throws BackendException {
+        super(configuration, DEFAULT_PORT);
+        this.keyspace = configuration.get(KEYSPACE);
+        this.atomicBatch = configuration.get(ATOMIC_BATCH_MUTATE);
+
+        this.cluster = initialiseCluster();
+        this.session = initialiseSession(this.cluster, this.keyspace);
+
+        final Configuration global = buildGraphConfiguration()
+                .set(READ_CONSISTENCY, "QUORUM")
+                .set(WRITE_CONSISTENCY, "QUORUM")
+                .set(METRICS_PREFIX, METRICS_SYSTEM_PREFIX_DEFAULT);
+
+        final Configuration local = buildGraphConfiguration()
+                .set(READ_CONSISTENCY, "LOCAL_QUORUM")
+                .set(WRITE_CONSISTENCY, "LOCAL_QUORUM")
+                .set(METRICS_PREFIX, METRICS_SYSTEM_PREFIX_DEFAULT);
+
+        final StandardStoreFeatures.Builder fb = new StandardStoreFeatures.Builder();
+
+        fb.batchMutation(true).distributed(true);
+        fb.timestamps(true).cellTTL(true);
+        fb.keyConsistent(global, local);
+        fb.multiQuery(true);
+
+        final String partitioner = this.cluster.getMetadata().getPartitioner();
+        switch (partitioner) {
+            case "org.apache.cassandra.dht.RandomPartitioner":
+            case "org.apache.cassandra.dht.Murmur3Partitioner": {
+                fb.keyOrdered(false).orderedScan(false).unorderedScan(true);
+                break;
+            }
+            case "org.apache.cassandra.dht.ByteOrderedPartitioner": {
```

Comment:
I don't think you want to have compile time dependencies on these classes as they are internals to cassandra and we really only want this depend on the driver jar and not the cassandra internals.
Perhaps it would be better to get the simplename of the class and switch on that accordingly?